web/src/app/pages/main-page-container/main-page/secondary-side-panel/secondary-side-panel.component.ts

```diff
@@ -28,18 +28,30 @@ import {
   styleUrls: ['./secondary-side-panel.component.css'],
 })
 export class SecondarySidePanelComponent {
+  subscription: Subscription = new Subscription();
+
   readonly sideNavMode = SideNavMode;
   readonly sideNavMode$: Observable<SideNavMode>;
   locationOfInterestId = '';
   submissionId = '';
 
   constructor(private navigationService: NavigationService) {
-    this.navigationService
-      .getLocationOfInterestId$()
-      .subscribe(id => (this.locationOfInterestId = id || ''));
-    this.navigationService
-      .getSubmissionId$()
-      .subscribe(id => (this.submissionId = id || ''));
+    this.subscription.add(
+      this.navigationService
+        .getLocationOfInterestId$()
+        .subscribe(id => (this.locationOfInterestId = id || ''))
+    );
+
+    this.subscription.add(
+      this.navigationService
+        .getSubmissionId$()
+        .subscribe(id => (this.submissionId = id || ''))
+    );
```

Comment:
The two streams feeding this component's state can receive events in any order, leading to race conditions. Instead, we should have a single subscription is called only once when the URL is changed. The easiest way to do this would be to expose `route.paramMap` from `NavigationService`, subscribe to that, and call `navigationService.getSubmissionId()` and `getLocationOfInterestId()` directly in the callback.